module-2/backend/week_three.md

```diff
@@ -10,16 +10,82 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What are the 3 main features in an ERD?
+  ```
+Entities, Relationships, Cardinality
+  ```
 2. Create an ERD for the following database: Restaurants, Customers, Items, Ingredients, Beverages, Orders, Vendors.
+```
+Restaurants => has_many => customers
+Restaurants => has_many => items 
+Items => has_many = ingredients 
+Vendors => has_many => Restaurants
+Restaurants => has_many => beverages
+Customer => has_many => orders
+```
 3. What is the entry at the command line to create a new rails app?
+```
+rails new appname
+```
 4. What do Models generally inherit from in rails?
+```
+Models inherit from Active Record
+```
 5. What do Controllers generally inherit from in a rails project?
+```
+Application Controller
+```
 6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+```
+there are a few ways i can go about it, but as far as the route goes:
+
+get '/horse', to: 'horses#show'
+resources :horses, only: [:show]
+
+def show
+  @horse = Horse.find(params[:id])
+end
+
+app/view/show.html.erb
+
+I think it kinda gives the idea? 
+```
 7. What rake task is useful when looking at routes, and what information does it give you?
+```
+rake routes, and it will give you the prefix, verb, uri, controller and action
+```
 8. What is an example of a route helper? When would you use them?
+```
+user_path or @user depending how you have your code set up.
+```
 9. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+```
+path starts from the '/' root and url gives the entire path including http....
+```
 10. What are strong params and why are the necessary?
+```
+Its my understanding that strong params limit the visibility and accessibility of information throghout the models, and it is applied for safety reasons.
+```
 11. What role does `form_for` play in helping us create our forms?
+```
+it communicates to the application what model the form belongs to, and help us gather the proper info from the user.
+```
 12. How does `form_for` know where to submit the user's input?
+```
+it essentially uses the controller to know what path to follow.
```

Comment:
it uses the object that we pass to it to determine where to submit the data - either `create` or `update` depending on if the object already exists in the database or not